repl/format.go

```diff
@@ -43,8 +43,10 @@ func FormatValue(v cty.Value, indent int) string {
 	case ty.IsPrimitiveType():
 		switch ty {
 		case cty.String:
-			// FIXME: If it's a multi-line string, better to render it using
-			// HEREDOC-style syntax.
+			// If this is a multi-line string, return it as-is
+			if strings.Contains(v.AsString(), "\n") {
+				return v.AsString()
+			}
```

Comment:
I'm not so sure about this. The goal of the using this formatter was to produce results which would round-trip to the same values. Treating multi-line strings like this just to suit the `type` function breaks this.

I do really like the multi-line output for the `type` result being a unformatted literal type. Could we achieve that some other way? Perhaps using [marks](https://github.com/zclconf/go-cty/blob/master/docs/marks.md), or [a capsule type](https://github.com/zclconf/go-cty/blob/master/docs/types.md#capsule-types)?

Reply:
I agree that this isn't the correct solution; when I get back to this PR I may take you up on the rubber duck offer since I'm curious how you envision marks/capsule type working in this case! :)

Reply:
🎉 @alisdair I see you already fixed this problem for me and it looks great! 😁 

```
> type(var.object)
<<EOT
object({
    "a": string,
    "b": object({
        "c": bool,
    }),
})

EOT
```